test/scholar/linear/linear_regression_test.exs

```diff
@@ -0,0 +1,567 @@
+defmodule Scholar.Linear.LinearRegressionTest do
+  use ExUnit.Case, async: true
+
+  describe "fit" do
+    test "matches sklearn for shapes {1, 1}, {1, 1} and type {:f, 32}" do
+      a = Nx.tensor([[0.5666993856430054]])
+      b = Nx.tensor([[0.8904717564582825]])
+      expected = Nx.tensor([[0.0]])
+      model = Scholar.Linear.LinearRegression.fit(a, b)
+      assert Nx.all_close(expected, model.coefficients) == Nx.tensor(1, type: {:u, 8})
+    end
+
+    test "matches sklearn for shapes {4, 6}, {4} and type {:f, 32}" do
+      a =
+        Nx.tensor([
+          [
+            0.9106853604316711,
+            0.2549760937690735,
+            0.6961511969566345,
+            0.9727275371551514,
+            0.21542717516422272,
+            0.1134142205119133
+          ],
+          [
+            0.06984847038984299,
+            0.5999580025672913,
+            0.5002026557922363,
+            0.3873320519924164,
+            0.4304788112640381,
+            0.7481111884117126
+          ],
+          [
+            0.8257377743721008,
+            0.3594258427619934,
+            0.08661065995693207,
+            0.5118331909179688,
+            0.38879409432411194,
+            0.3640798032283783
+          ],
+          [
+            0.060781270265579224,
+            0.054385241121053696,
+            0.6188914775848389,
+            0.9003549218177795,
+            0.7764868140220642,
+            0.9584161043167114
+          ]
+        ])
+
+      b =
+        Nx.tensor([0.608439028263092, 0.6562057137489319, 0.9454836249351501, 0.8614323735237122])
+
+      expected =
+        Nx.tensor([
+          0.14827239513397217,
+          -0.2630932927131653,
+          -0.45938295125961304,
+          -0.02042902633547783,
+          0.2715783417224884,
+          0.16200493276119232
+        ])
+
+      %Scholar.Linear.LinearRegression{coefficients: actual} =
+        Scholar.Linear.LinearRegression.fit(a, b)
+
+      assert Nx.all_close(expected, actual, rtol: 1.0e-2, atol: 1.0e-2) ==
+               Nx.tensor(1, type: {:u, 8})
```

Comment:
I'd bring the `assert_all_close` helper from Nx into Scholar. Helps a lot with these assertions